File: frontend/lib/core/atomic_objects/pageable.dart
class Pageable {
  int totalItems;
  int itemsPerPage;
  int page;
  String sorting;

  Pageable(
      {required this.totalItems,
      required this.itemsPerPage,
      required this.page,
      required this.sorting});

  factory Pageable.fromJson(Map<String, dynamic> json) {
    return Pageable(
      totalItems: json['totalItems'],
      itemsPerPage: json['itemsPerPage'],
      page: json['page'],
      sorting: json['sorting'],
    );
  }
}

File: backend/src/main/java/de/fraunhofer/DigiTales2Go/dataStructure/dataStructureSubclasses/Pageable/Pageable.java
package de.fraunhofer.DigiTales2Go.dataStructure.dataStructureSubclasses.Pageable;

import de.fraunhofer.DigiTales2Go.dataStructure.dataStructureSubclasses.enumerations.Sorting;
import lombok.Data;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;

@Data
public class Pageable {
    private int totalItems =0;
    @Min(1)
    @NotNull
    private int itemsPerPage;
    @Min(0)
    @NotNull
    private int page;

    private Sorting sorting = Sorting.NOTSET;

}
